### DISCUSSION TITLE
Does Sketchybar work without yabai?

### DISCUSSION THREAD

**Discussion by bassamsdata** [FelixKratz/SketchyBar#287] [Q&A] [answered]
Since I'm using Amethyst instead of Yabai, does Sketchybar work without Yabai? 
Particularly I'm interested in **spaces**, because I read some scripts and found that they rely on Yabai.

As I couldn't find an answer in the Q&A, I posted this question here.
As a note, I use Amethyst and karabiner.

I appreciate your help and Thanks a lot in advance.

**Comment by FelixKratz** [comment, marked as answer]
Yes, sketchybar works without yabai. Some of the scripts posted in the plugins discussion use yabai functionality but it is not needed generally. For example the default config doesn't use yabai for anything except for focusing spaces on click, but that can be changed as well. Some events from yabai can be helpful for advanced plugins however.

**Reply by bassamsdata** [reply]
Thank you for your answer 🙏. I'll go ahead and install it.

I was intrigued by your config of the spaces in Sketchybar. Can that be achieved without yabai?

**Reply by FelixKratz** [reply]
> Thank you for your answer 🙏. I'll go ahead and install it.
> 
> I was intrigued by your config of the spaces in Sketchybar. Can that be achieved without yabai?

You would need some other program that knows about all application windows on all spaces. I dont know if there are other programs supplying this information. The program that would be predetermined to deliver these kind if information is the window manager, so maybe amethyst has some api to get these informations? Since you don't want to poll these informations (for performance and latency reasons) the program would also need to be able to deliver events on a change in those informations.

Maybe I will work on making something like this possible via sketchybar directly… not sure yet

**Reply by bassamsdata** [reply]
Yeah, I was expecting that. Because I haven't seen any article or so about Amethyst providing such APIs, I don't think it does, but in reality I don't know.

I'm going to try Sketchybar with Ametheyst and Karabiner, and I'm sure I'll go down a rabbit hole for a week trying to figure things out 🤓. 
Anyway, I'll share my experience with Amethyst and Sketchybar. I hope I don't end up using Yabai 😂(actually I've avoided installing it for quite some time).

Sketchybar is a great tool with its current settings, and not having to rely on other apps especially for this feature would be an xtra xtra awesome, but I know it would require a great deal of work. And I'm not sure how many would benefit, since most people use Yabai.

**Reply by meksrc** [reply]
Hey :) before I try to install this, did you get it to work?
I am also using Amethyst, would it be possible to get sketchy bar running without these informations? I would just like to see how many spaces I have, thats it.

**Reply by bassamsdata** [reply]
Yea it's working perfectly, I have no issue either with spaces or another plugins.
here is a video to illustrate that:

https://user-images.githubusercontent.com/105807570/215943995-989ad035-046f-431a-b288-54c07ff9b041.mov

**Comment by meksrc** [comment]
Oh I got it working tysm!

I tried it months ago but somehow I missed the instructions right after install.
Maybe the creator of this plugin could add these to the GitHub page?

**Reply by FelixKratz** [reply]
The problem is that the commands listed after the brew install are specific to the macOS version and brew version, so it is not possible to put them on the website. The instructions contain a disclaimer after the install command on the website.
